astropy/cosmology/_io/yaml.py

```diff
@@ -50,19 +73,7 @@ def yaml_representer(tag):
         Function to construct :mod:`yaml` representation of |Cosmology| object.
     """
 
-    def representer(dumper, obj):
-        """Cosmology yaml representer function for {}.
```

Comment:
I don't think I get why this docstring was moved. It's only ever used once so what's the expected benefit of moving it away from the code ?

Reply:
static type checkers don't like modifying existing docs. Python type hints docs as `str | None` and type checkers can't figure out that it's definitely a `str` even if it was just assigned. Therefore it's better to do it in 1 go and then assign the docs.